packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1002,6 +1034,112 @@ class _SliverAppBarDelegate extends SliverPersistentHeaderDelegate {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=freeform}
+///
+/// ```dart imports
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart
+/// void main() => runApp(MyApp());
+///
+/// class MyApp extends StatefulWidget {
+///   const MyApp({Key key}) : super(key: key);
+///
+///   @override
+/// State<StatefulWidget> createState() => _MyAppState();
+/// }
+/// class _MyAppState extends State<MyApp> {
+///   bool _pinned = true;
+///   bool _snap = false;
+///   bool _floating = false;
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return MaterialApp(
+///       home: Scaffold(
+///       // SliverAppBar is declared in Scaffold.body, in slivers of a
```

Comment:
Please make all comments before the example.

Reply:
ok and also thanks for review .